Nodejs/Product/TestAdapter/TestFrameworks/Jest/jest.js

```diff
@@ -0,0 +1,156 @@
+﻿// @ts-check
+"use strict";
+const fs = require('fs');
+const os = require('os');
+const path = require('path');
+
+const find_tests = function (testFileList, discoverResultFile, projectFolder) {
+    // Jest package is not required for this tests to run, but it's still good to check 
+    // in case the user have a build system or an automation that requires Jest.
+    const jest = detectPackage(projectFolder, 'jest');
+    const jestEditorSupport = detectPackage(projectFolder, 'jest-editor-support');
+
+    if (!jest || !jestEditorSupport) {
+        return;
+    }
+
+    let testList = [];
+    for (let testFile of testFileList.split(';')) {
+        process.chdir(path.dirname(testFile));
+
+        try {
+            const parseResult = jestEditorSupport.parse(testFile);
+
+            for (let test of parseResult.itBlocks) {
+                testList.push({
+                    column: test.start.column,
+                    file: test.file,
+                    line: test.start.line,
+                    test: test.name
+                });
+            }
+
+        }
+        catch (e) {
+            // We would like continue discover other files, so swallow, log and continue;
+            console.error("Test discovery error:", e, "in", testFile);
+        }
+    }
+
+    const fd = fs.openSync(discoverResultFile, 'w');
+    fs.writeSync(fd, JSON.stringify(testList));
+    fs.closeSync(fd);
+};
+
+const run_tests = function (testCases, post) {
+    // Jest package is not required for this tests to run, but it's still good to check 
+    // in case the user have a build system or an automation that requires Jest.
+    const jest = detectPackage(testCases[0].projectFolder, 'jest');
+    const jestEditorSupport = detectPackage(testCases[0].projectFolder, 'jest-editor-support');
+
+    if (!jest || !jestEditorSupport) {
+        return;
+    }
+
+    const workspace = {
+        rootPath: testCases[0].projectFolder,
+        pathToJest: getJestPath(testCases[0].projectFolder),
+        pathToConfig: '',
+        localJestMajorVersion: 23 // TODO: Get the jest version from the package.
+    };
+
+    runTest(testCases)
+        .then(() => post({
+            type: 'suite end',
+            result: {}
+        }))
+        .catch((error) => logError(error));
+
+    function runTest(testCases) {
+        return new Promise((resolve, reject) => {
+
+            // Start all test cases. Jest doesn't report when each test start, only reports the whole file.
```

Comment:
So do tests now stream on a per-file basis, or do the results for the entire suite just come back at once?

Reply:
Per-file basis. I decided not to change that yet, as the change is a little bit bigger than this one. In any case if there's a performance issue we can improve that portion as well.